clients/resolution/pypi_registry_client.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resolution
+
+import (
+	"context"
+
+	"deps.dev/util/pypi"
+	"deps.dev/util/resolve"
+	"deps.dev/util/resolve/dep"
+	"github.com/google/osv-scalibr/clients/datasource"
+)
+
+// PyPIRegistryClient is a client to fetch data from PyPI registry.
+type PyPIRegistryClient struct {
+	api *datasource.PyPIRegistryAPIClient
+}
+
+// NewPyPIRegistryClient makes a new PyPIRegistryClient.
+func NewPyPIRegistryClient(registry string) *PyPIRegistryClient {
+	return &PyPIRegistryClient{api: datasource.NewPyPIRegistryAPIClient(registry)}
+}
+
+// Version returns metadata of a version specified by the VersionKey.
+func (c *PyPIRegistryClient) Version(ctx context.Context, vk resolve.VersionKey) (resolve.Version, error) {
+	// Version is not used by the PyPI resolver for now, so here only returns the VersionKey.
+	// We may need to add more metadata in the future.
+	return resolve.Version{VersionKey: vk}, nil
+}
+
+// Versions returns all the available versions of the package specified by the given PackageKey.
+func (c *PyPIRegistryClient) Versions(ctx context.Context, pk resolve.PackageKey) ([]resolve.Version, error) {
+	pkgInfo, err := c.api.GetPackageInfo(ctx, pk.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	var vks []resolve.Version
+	for v := range pkgInfo.Releases {
```

Comment:
Does the resolver expect this to be in order? You might need to sort this because of go maps.